### PR TITLE
Upmerge 1.12 into 1.13

### PR DESCRIPTION
**Pull request by jakubtobiasz**
Upmerge 1.12 into 1.13

### CODE REVIEW

File: src/Sylius/Bundle/CoreBundle/CatalogPromotion/Processor/CatalogPromotionRemovalProcessor.php [hunk 1]
```diff
@@ -28,7 +28,7 @@ public function __construct(
         private CatalogPromotionRepositoryInterface $catalogPromotionRepository,
         /** @var CatalogPromotionRemovalAnnouncerInterface $catalogPromotionRemovalAnnouncer */
         private CatalogPromotionRemovalAnnouncerInterface|MessageBusInterface $catalogPromotionRemovalAnnouncer,
-        private ?MessageBusInterface $eventBus = null,
+        private ?MessageBusInterface $eventBus = null, /** @phpstan-ignore-line */
```
Comment (jakubtobiasz) [inline src/Sylius/Bundle/CoreBundle/CatalogPromotion/Processor/CatalogPromotionRemovalProcessor.php:31]: As the below is triggered a deprecation, I decided to leave it with ignore.

File: src/Sylius/Bundle/ApiBundle/EventSubscriber/TaxonDeletionEventSubscriber.php [hunk 2]
```diff
@@ -21,13 +21,11 @@
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpKernel\Event\ViewEvent;
 use Symfony\Component\HttpKernel\KernelEvents;
-use Symfony\Component\Messenger\MessageBusInterface;
 
 /** @experimental */
 final class TaxonDeletionEventSubscriber implements EventSubscriberInterface
 {
     public function __construct(
-        private MessageBusInterface $eventBus,
```
Comment (jakubtobiasz) [inline src/Sylius/Bundle/ApiBundle/EventSubscriber/TaxonDeletionEventSubscriber.php:30]: Removed as listeners are excluded from BC and it's safe.

File: src/Sylius/Bundle/AdminBundle/Controller/DashboardController.php [hunk 3]
```diff
@@ -31,7 +30,6 @@ public function __construct(
         private ChannelRepositoryInterface $channelRepository,
         private Environment $templatingEngine,
         private RouterInterface $router,
-        private ?SalesDataProviderInterface $salesDataProvider = null,
```
Comment (jakubtobiasz) [inline src/Sylius/Bundle/AdminBundle/Controller/DashboardController.php:34]: Removed as controllers are excluded from BC and there's no previous deprecation triggering.

File: src/Sylius/Bundle/PayumBundle/Controller/PayumController.php [hunk 4]
```diff
@@ -43,7 +42,6 @@ public function __construct(
         private OrderRepositoryInterface $orderRepository,
         private MetadataInterface $orderMetadata,
         private RequestConfigurationFactoryInterface $requestConfigurationFactory,
-        private ViewHandlerInterface $viewHandler,
```
Comment (jakubtobiasz) [inline src/Sylius/Bundle/PayumBundle/Controller/PayumController.php:46]: Removed as controllers are excluded from BC and there's no previous deprecation triggering.

**Review by Rafikooo — APPROVED** [review]

**Pull request merged**